Library/TelemetryClient.ts

```diff
@@ -118,7 +118,14 @@ class TelemetryClient {
             // url.parse().host returns null for non-urls,
             // making this essentially a no-op in those cases
             // If this logic is moved, update jsdoc in DependencyTelemetry.target
-            telemetry.target = url.parse(telemetry.data).host;
+            // url.parse() is deprecated, update to use WHATWG URL API instead
+            try {
+                telemetry.target = new url.URL(telemetry.data).host;
```

Comment:
A `TypeError `will be thrown if the input or base are not valid URLs. 
To be compliant with previous behavior, when url is invalid, set `telemetry.target` as `null`.